starter-code/src/functions-and-arrays.js

```diff
@@ -1,100 +1,180 @@
 // Find the maximum
+function maxOfTwoNumbers(a, b) {
+	if (a > b) {
+		return a;
+	} else if (a < b) {
+		return b;
+	} else {
+		return a, b;
+	}
+}
+console.log(maxOfTwoNumbers(3, 2));
 
 // Finding Longest Word
-var words = [
-  'mystery',
-  'brother',
-  'aviator',
-  'crocodile',
-  'pearl',
-  'orchard',
-  'crackpot'
-];
+var words = [ 'mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot' ];
+
+function findLongestWord(arr) {
+	var word = [ '0' ];
+	if (arr.length === 0) {
+		return undefined;
+	} else {
+		for (let i = 0; i < arr.length; i++) {
+			if (arr[i].length > word[0].length) {
+				word.splice(0, 1, arr[i]);
+			}
+		}
+	}
+	return word[0];
+}
+
+console.log(findLongestWord(words));
 
 // Calculating a Sum
 
-var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+var numbers = [ 6, 12, 1, 18, 13, 16, 2, 1, 8, 10 ];
+
+function sumArray(arr) {
+	var sumator = 0;
+	if (arr.length === 0) {
+		return 0;
+	} else {
+		for (let i = 0; i < arr.length; i++) {
+			sumator += arr[i];
+		}
+	}
+	if (sumator === 0) {
+		return 0;
+	} else {
+		return sumator;
+	}
+}
+
+console.log(sumArray(numbers));
 
 // Calculate the Average
 
-var numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+var numbersAvg = [ 2, 6, 9, 10, 7, 4, 1, 9 ];
+
+function averageNumbers(arr) {
+	var sumator = 0;
+	if (arr.length === 0) {
+		return undefined;
+	} else {
+		for (let i = 0; i < arr.length; i++) {
+			sumator += arr[i];
+		}
+	}
+	return sumator / arr.length;
+}
+
+console.log(averageNumbers(numbersAvg));
 
 // Array of Strings
-var wordsArr = [
-  'seat',
-  'correspond',
-  'linen',
-  'motif',
-  'hole',
-  'smell',
-  'smart',
-  'chaos',
-  'fuel',
-  'palace'
-];
+var wordsArr = [ 'seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace' ];
+
+function averageWordLength(arr) {
+	var sumator = 0;
+	if (arr.length === 0) {
+		return undefined;
+	} else {
+		for (let i = 0; i < arr.length; i++) {
```

Comment:
while this works, using a reduce would be better. A reduce is a really powerful array method, but it also signified intent to other people who are reading the code. If they see a reduce, they know what you're trying to do without reading through the loop